### PR TITLE
Add notes

### LINKED ISSUE #68: Create backend data sctructure
Ad decided in a meeting with @tomimarkus991 @Kasparsu @Erikyy @tuulingo this is the basic data structure for the project:
```sh
User: 
* username
* tel. nr
* email
* pw
* reg-date
* role (ENUM "ADMIN", "USER")
* full name
* personal-code?
- Loan[]

Category # electroncs/fildming/cards/200-400gb/sony-mega-card-2000
* name
* Bundelable (bool)
- Product[]
- Category[]

Product: # unique product (e.g. sony fx-9)
* name
* description
* Specs json{}
- Category[]
- items[]

Item: # Instance of product (e.g. lightly scratched sony fx-9)
* reg-code
- product
- Notes[] # Latest one is the current state (older ones for history)
- Loan[]

Note:
* content
* Item

Bundle: # Set of products for ease of use (e.g. sony filming setup with a bunch of stuff)
* description
- Product[]
- Category[]

Loan: # 
* User
* message
- Bundle
- LoanItem[]

LoanItem
* Item
* returnedAt (date)
- LoanRequest[]

LoanRequest
* User
* Category
* createdAt (date)
* Request (ENUM: "extend", "request-loan", "return")
- item (LoanItem)
- loan (Loan)
- LoanDecion

LoanDecision:
* from (date)
* to (date)
* Decider (user)
* Decision (ENUM: "approve", "deny")
```

### 1

- [ ] User
- [ ] Category
- [ ] Note
- [ ] Item

### 2

- [ ] Product
- [ ] Bundle
- [ ] Loan
- [ ] LoanItem

### 3 

- [ ] LoanRequest
- [ ] LoanDecision

### Please create CRUD actions for all the items.

@TPT-Loane/back-end, grab any of the items to yourself by tagging your @ behind the item.
Please also create WIP pr-s right away and push to them semi regularly to maintain visibility.

Let's name everything **singular**. (both db and nest logic). Only thing plural should be the gql queries which return many things. 

Nest structure

```
user.module
- User logic
Item.module
- Everything that has to do with items
loan.module
- Everything that has to do with loan logic
```

### PR DESCRIPTION
**Pull request by tuulingo**
## Header
Add notes #68 

## Description

- [] Add operations for notes (create, read, delete). Didn't include update, because we don't want it.
- [] Commented "Item" structure will be fixed in another branch

Fixes #68 notes data structure

## Branch name

Add-notes

### CODE REVIEW

File: be/src/note/dto/update-note.input.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { InputType, Int, Field } from '@nestjs/graphql';
+import { Column } from 'typeorm';
+
+@InputType()
+export class UpdateNoteInput {
```
Comment (Orzelius) [inline be/src/note/dto/update-note.input.ts:5]: The `UpdateNoteInput` and `CreateNoteInput` classes are the same currently. Might as well have only one for now

File: be/src/note/entities/note.entity.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@ObjectType()
+@Entity('items')
+export class Note {
+
+    @Field(() => Int)
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Field()
+    @Column({ length: 255, nullable: false })
```
Comment (Orzelius) [inline be/src/note/entities/note.entity.ts:13]: This length might be not enough. Some people might want to have lengthy notes 🤷

File: be/src/note/note.service.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    return this.notesRepository.save(updateNoteInput);
+  }
+
+  remove(id: number) {
+    return this.notesRepository.delete(id);
+  }
+
+  //  getLoans(loanId: number): Promise<Loans> {
+  //    return this.loansService.findOne(loanId)
+  //  }
```
Comment (Orzelius) [inline be/src/note/note.service.ts:36]: Some leftover code? Plz remove

**Review by Orzelius — CHANGES_REQUESTED** [review]
Some small changes would be nice.

This pr was structured a bit weird. You should had named the fields `note` straight away and not had a commit that changed the names. Also Let's commit all gen code (schema, migrations) separately as there's no point in reviewing it.

**Review by Orzelius — APPROVED** [review]

File: be/src/note/dto/create-note.input.ts [hunk 4]
```diff
@@ -4,6 +4,6 @@ import { Column } from 'typeorm';
 @InputType()
 export class CreateNoteInput {
     @Field()
-    @Column({ length: 255, nullable: false })
+    @Column({ nullable: false })
```
Comment (Orzelius) [inline be/src/note/dto/create-note.input.ts:7]: Oh wait, I did a lazy review, this shouldn't be here. the `@Column` decorator come from `typeorm` package and has nothing to do with gql input object. `@Column` is used in typeorm entities do describe db structure. This class here describes the input needed to gql to create a new Note.

**Review by Orzelius — CHANGES_REQUESTED** [review]

**Review by Orzelius — APPROVED** [review]

File: be/src/note/dto/create-note.input.ts (hunk 4 above)
Comment (Orzelius) [reply be/src/note/dto/create-note.input.ts]: Fixed -> 6066234

**Comment by Orzelius** [comment]
@tuulingo fyi when fixing things up from reviews then it's easier for the reviewer if separated each review request change into separate commit (Like you did with 6066234) and comment to the review-comment what you changed with the commit hash (like [this](https://github.com/TPT-Loane/TPT-Loane/pull/88#discussion_r632419235))

**Review by tomimarkus991 — APPROVED** [review]

**Review by Erikyy — APPROVED** [review]

**Pull request merged**